### PR TITLE
feat(dew): add a new resource to manage secret version state

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a new resource to manage secret version state.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support a new resource to  manage secret version state.
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccSecretVersionState_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccSecretVersionState_basic -timeout 360m -parallel 4
=== RUN   TestAccSecretVersionState_basic
=== PAUSE TestAccSecretVersionState_basic
=== CONT  TestAccSecretVersionState_basic
--- PASS: TestAccSecretVersionState_basic (52.88s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       52.928s
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<
image
    ab. Related resources (parent resources) not found
    \>>>>>> Paste the screenshot here <<<<<<
image

  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<
image

### CODE REVIEW

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret_version_state.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	stateId, err := jmespath.Search("stage.name", createVersionStateRespBody)
+	if err != nil || stateId == nil {
+		return diag.Errorf("error creating secret version state: ID is not found in API response")
+	}
+
+	d.SetId(stateId.(string))
```
Comment (luoping-12345) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret_version_state.go]:
```suggestion
	stateName := utils.PathSearch("stage.name", createVersionStateRespBody, "").(string)
	if stateName == "" {
		return diag.Errorf("error creating secret version state: stage.name is not found in API response")
	}

	d.SetId(stateName)
```

File: docs/resources/csms_secret_version_state.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+## Example Usage
+
+```hcl
+variable "secret_name" {}
+variable "name" {}
+variable "version_id" {}
+
+resource "huaweicloud_csms_secret_version_state" "test" {
+  secret_name = var.secret_name
+  name        = var.secret_name
```
Comment (jinyangyang222) [inline docs/resources/csms_secret_version_state.md:28]:
```suggestion
  name        = var.name
```

File: huaweicloud/services/acceptance/dew/resource_huaweicloud_csms_secret_version_state_test.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSecretVersionState_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "secret_name", "huaweicloud_csms_secret.test", "name"),
+					resource.TestCheckResourceAttr(rName, "name", name),
+					resource.TestCheckResourceAttrPair(rName, "version_id", "huaweicloud_csms_secret.test", "latest_version"),
+					resource.TestCheckResourceAttrSet(rName, "updated_at"),
```
Comment (jinyangyang222) [inline huaweicloud/services/acceptance/dew/resource_huaweicloud_csms_secret_version_state_test.go:70]: Please use RFC3339 format for verification

File: huaweicloud/services/acceptance/dew/resource_huaweicloud_csms_secret_version_state_test.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+	}
+}
+
+func testAccSecretVersionState_basic(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_csms_secret" "test" {
+  name        = "%[1]s"
+  secret_text = "tf"
+  description = "acc test"
+}
```
Comment (jinyangyang222) [inline huaweicloud/services/acceptance/dew/resource_huaweicloud_csms_secret_version_state_test.go:116]: This resource can be extracted and reused as a base function（testAccSecretVersionState_base）

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret_version_state.go [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+	}
+
+	createVersionStateRespBody, err := utils.FlattenResponse(creatVersionStateResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	stateName := utils.PathSearch("stage.name", createVersionStateRespBody, "").(string)
+	if stateName == "" {
+		return diag.Errorf("error creating secret version state: the stage.name is not found in API response")
```
Comment (jinyangyang222) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret_version_state.go:96]:
```suggestion
		return diag.Errorf("error creating secret version state: the name is not found in API response")
```

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret_version_state.go [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+	deleteVersionStatePath = strings.ReplaceAll(deleteVersionStatePath, "{stage_name}", d.Id())
+	deleteVersionStateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			204,
+		},
+	}
+
+	_, err = client.Request("DELETE", deleteVersionStatePath, &deleteVersionStateOpt)
+	if err != nil {
```
Comment (jinyangyang222) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret_version_state.go:210]: Delete function need add checkDeleted logic

File: docs/resources/csms_secret_version_state.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+* `id` - The resource ID.
+
+* `updated_at` - The last update time of the secret version state, in RFC3339 format.
+
+## Import
+
+The secret version state can be imported using the related `secret_name` and their `id`, separated by a slash (/), e.g.
+
+```bash
+terraform import huaweicloud_csms_secret_version_state.test <secret_name>/<id>
```
Comment (jinyangyang222) [inline docs/resources/csms_secret_version_state.md:63]:
```suggestion
terraform import huaweicloud_csms_secret_version_state.test <secret_name>/<name>
```

File: docs/resources/csms_secret_version_state.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+* `updated_at` - The last update time of the secret version state, in RFC3339 format.
+
+## Import
+
+The secret version state can be imported using the related `secret_name` and their `id`, separated by a slash (/), e.g.
```
Comment (jinyangyang222) [inline docs/resources/csms_secret_version_state.md:60]:
```suggestion
The secret version state can be imported using the related `secret_name` and their `name`, separated by a slash (/), e.g.
```

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/dew" TESTARGS="-run TestAccSecretVersionState_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccSecretVersionState_basic -timeout 360m -parallel 4
=== RUN   TestAccSecretVersionState_basic
=== PAUSE TestAccSecretVersionState_basic
=== CONT  TestAccSecretVersionState_basic
--- PASS: TestAccSecretVersionState_basic (53.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       53.814s
```

File: docs/resources/csms_secret_version_state.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  Only letters, digits, underscores(_) and hyphens(-) are allowed.
+  The valid length is limited from `1` to `64` characters.
+
+* `version_id` - (Required, String) Specifies the ID of the secret version.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID, set `name` as the ID.
```
Comment (jinyangyang222) [inline docs/resources/csms_secret_version_state.md:54]:
```suggestion
* `id` - The resource ID, same as `name`.
```

**Pull request merged**